### PR TITLE
Improve reference.copy() so that it is possible to override attributes.

### PR DESCRIPTION
**Pull request by frankban**
This just enables some sugar for callers.

### CODE REVIEW

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-bundlelib/59/
Test PASSed.

File: jujubundlelib/references.py [hunk 1]
```diff
@@ -161,11 +161,23 @@ def similar(self, other):
...
+
+            ref = reference.copy()
+            ref = reference.copy(revision=42)
+            ref = reference.copy(channel='', user='')
+        """
+        reference = self.__class__(
             self.schema, self.user, self.channel, self.series, self.name,
             self.revision)
+        for key, value in kwargs.items():
+            setattr(reference, key, value)
```
Comment (fabricematrat) [inline jujubundlelib/references.py]: May be   
 try:
        setattr(reference, key, value)
    except AttributeError:
        pass or log

??

**Comment by fabricematrat** [comment]
:+1:

**Comment by frankban** [comment]
Thanks for the review!
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-bundlelib-merge

**Pull request merged**